README.rst

```diff
@@ -79,10 +79,14 @@ https://numba.readthedocs.io/en/stable/index.html
 Mailing Lists
 =============
 
-Join the Numba mailing list numba-users@continuum.io:
-https://groups.google.com/a/continuum.io/d/forum/numba-users
```

Comment:
oh... didn't kn we still have that link here

Reply:
Only just realised today!